### PR TITLE
Cart list _ wishlist

### PR DESCRIPTION
**Pull request by Tanjum786**
Added cart and wishlist functionality to the project :

- added ADD TO CART 
- added ADD TO WISHLIST
- Added REMOVE FROM WISHLIST
- added REMOVE FROM CART 
- added MOVE TO WISH LIST
- added MOVE TO CART
- added Increase and Decrease quantity 

https://user-images.githubusercontent.com/84380051/162582067-edcc68fd-060c-4870-8f5a-9682c4f8357c.mp4

### CODE REVIEW

File: src/pages/ProductLists/productpage.css [hunk 1]
```diff
@@ -159,3 +159,11 @@
     color: var(--secondary-color);
     margin: 1rem;
 }
+.wishlist_btn{
+    border: none;
+    font-size: 1rem;
+    background: none;
+}
+.color-btn{
+    color: red;
```
Comment (vaibhav-mahalle) [inline src/pages/ProductLists/productpage.css]: try and define color variables and then use them.

File: src/pages/Wishlist/component/Wishlistcard.jsx [hunk 2]
```diff
@@ -1,27 +1,65 @@
...
+                        rating: rating,
+                        Quantity: Quantity,
+                      },
+                    })
+                  }
+                >
+                  <i className="fas fa-heart"></i>
+                </button>
+              </div>
+              <img src={img} alt="img" />
```
Comment (vaibhav-mahalle) [inline src/pages/Wishlist/component/Wishlistcard.jsx]: naming for alt tag could be better.

**Review by vaibhav-mahalle — APPROVED** [review]
suggested some changes please look into it.

File: src/hooks/context/Reducer/CartRedcerFun.jsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
+export const CartRedcerFun = (cartstate, cartaction) => {
```
Comment (SimranjitKaurMaan) [inline src/hooks/context/Reducer/CartRedcerFun.jsx]: Check the Typo

Comment (SimranjitKaurMaan) [reply src/hooks/context/Reducer/CartRedcerFun.jsx]: You can rename it as `CartReducer`. Func is redundant.

File: src/pages/Cart/component/Cartbill.jsx [hunk 4]
```diff
@@ -1,41 +1,44 @@
...
-  price,
-  cartitem,
-  DiscAmt,
-  DeliveryAmt,
-  TotalAmt,
-  SaveAmt,
-}) => {
+import { useProducts } from "../../../hooks/context/cart-wishlistcontext";
+
+const Cartbill = () => {
```
Comment (SimranjitKaurMaan) [inline src/pages/Cart/component/Cartbill.jsx:3]: First letter of a component should be UpperCase.

File: src/pages/Wishlist/Wishlist.jsx [hunk 5]
```diff
@@ -1,55 +1,43 @@
...
-           <Wishlistcard
-            cardimg={image7}
-            cardAlt={"image"}
-            cardPrice={"Rs.2500"}
-            cardTitle={"Kid's Festive & party ware"}
-            cardDic={"FTC Fashions"}
-          /> */}
-        </section>
+        <Navbar auth={"Login"} linkto={"/login"} />
+        {mywishlist.length <= 0 ? (
```
Comment (SimranjitKaurMaan) [inline src/pages/Wishlist/Wishlist.jsx]: Avoid using myWishList naming. Use something like wishList.

**Review by SimranjitKaurMaan — APPROVED** [review]
Suggested few things. Good to merge.

**Pull request merged**